sql/sql_list.h

```diff
@@ -168,11 +176,19 @@ class base_list :public Sql_alloc
     relies on this behaviour. This logic is quite tricky: please do not use
     it in any new code.
   */
-  inline base_list(const base_list &tmp) :Sql_alloc()
+  base_list(const base_list &tmp)
+    : Sql_alloc(),
+      first(tmp.first),
+      last(tmp.elements ? tmp.last : &first),
+      elements(tmp.elements)
+  {
+  }
+  base_list &operator=(const base_list &tmp)
   {
     elements= tmp.elements;
```

Comment:
I would also add base class assignment operator call
`Sql_alloc::operator = (*this);`

Reply:
done